assets/src/hooks/useVehiclesSocket.ts

```diff
@@ -0,0 +1,175 @@
+import { Channel, Socket } from "phoenix"
+import { Dispatch as ReactDispatch, useEffect, useReducer } from "react"
+import { ByRouteId, RouteId, Vehicle, VehiclesByRouteId } from "../skate.d"
+
+interface State {
+  socket: Socket | undefined
+  channelsByRouteId: ByRouteId<Channel>
+  vehiclesByRouteId: VehiclesByRouteId
+}
+
+export const initialState: State = {
+  socket: undefined,
+  channelsByRouteId: {},
+  vehiclesByRouteId: {},
+}
+
+interface SetSocketAction {
+  type: "SET_SOCKET"
+  payload: {
+    socket: Socket
+  }
+}
+
+export const setSocket = (socket: Socket): SetSocketAction => ({
+  type: "SET_SOCKET",
+  payload: { socket },
+})
+
+interface SetChannelForRouteAction {
+  type: "SET_CHANNEL"
+  payload: {
+    routeId: RouteId
+    channel: Channel
+  }
+}
+
+export const setChannelForRoute = (
+  routeId: RouteId,
+  channel: Channel
+): SetChannelForRouteAction => ({
+  type: "SET_CHANNEL",
+  payload: { routeId, channel },
+})
+
+interface SetVehiclesForRouteAction {
+  type: "SET_VEHICLES_FOR_ROUTE"
+  payload: { routeId: RouteId; vehicles: Vehicle[] }
+}
+
+export const setVehiclesForRoute = (
+  routeId: RouteId,
+  vehicles: Vehicle[]
+): SetVehiclesForRouteAction => ({
+  type: "SET_VEHICLES_FOR_ROUTE",
+  payload: { routeId, vehicles },
+})
+
+interface RemoveRouteAction {
+  type: "REMOVE_ROUTE"
+  payload: {
+    routeId: RouteId
+  }
+}
+
+export const removeRoute = (routeId: RouteId): RemoveRouteAction => ({
+  type: "REMOVE_ROUTE",
+  payload: { routeId },
+})
+
+type Action =
+  | SetSocketAction
+  | SetChannelForRouteAction
+  | SetVehiclesForRouteAction
+  | RemoveRouteAction
+
+type Dispatch = ReactDispatch<Action>
+
+export const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case "SET_SOCKET":
+      return {
+        ...state,
+        socket: action.payload.socket,
+      }
+    case "SET_CHANNEL":
+      return {
+        ...state,
+        channelsByRouteId: {
+          ...state.channelsByRouteId,
+          [action.payload.routeId]: action.payload.channel,
+        },
+      }
+    case "SET_VEHICLES_FOR_ROUTE":
+      return {
+        ...state,
+        vehiclesByRouteId: {
+          ...state.vehiclesByRouteId,
+          [action.payload.routeId]: action.payload.vehicles,
+        },
+      }
+    case "REMOVE_ROUTE":
+      const {
+        [action.payload.routeId]: channel,
```

Comment:
Completely a style thing, but naming it `_ channel` (and `_ vehicles` below) would indicate that you aren't interested in using the value of this variable.